### PR TITLE
w03 d03 mongoose reference lesson

### PR DESCRIPTION
**Pull request by bgveenstra**
- combined readmes from past classes
- added generic, simplified exercises (could continue earlier work or start anew)
- TODO: solutions

untested; ready for a grammar / formatting / sanity check review

### CODE REVIEW

File: 02-express-mongo-crud/mongoose-reference-populate/27-28/README.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+```
+
+Notice that we start the `consoles` array empty within our zelda game document. That will be filled with ObjectIds later on.
+
+Now we'll save our work.
+
+```js
+nin64.save(function(err, nintendo64) {
+ if(err) {return console.error(err);}
+ else console.log('nintendo 64 saved successfully');
```
Comment (tgaff) [inline 02-express-mongo-crud/mongoose-reference-populate/27-28/README.md]: You might want to change this syntax and the spacing above.  It's not style-guide compliant.

File: 02-express-mongo-crud/mongoose-reference-populate/27-28/README.md [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+});
+
+zelda.consoles.push(nin64);
+zelda.save(function(err, zeldaGame) {
+ if(err) {return console.error(err);}
```
Comment (tgaff) [inline 02-express-mongo-crud/mongoose-reference-populate/27-28/README.md]: ditto

File: 02-express-mongo-crud/mongoose-reference-populate/27-28/README.md [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+      if (err) { res.status(500).send(err); }
+      console.log('found and populated game: ', game);
+      res.json(game);
+    });
+});
+```
+
+```js
+// send all information for all games
+app.get('/api/games/', function (req, res) {
```
Comment (tgaff) [inline 02-express-mongo-crud/mongoose-reference-populate/27-28/README.md:369]: Maybe add some discussion that for an index route you might NOT send all this data.

**Comment by tgaff** [comment]
Also you may not need exercises if we're using the lab

**Comment by bgveenstra** [comment]
Thanks! I'm planning to remove exercises, and will fix styleguide.

**Pull request merged**